### PR TITLE
Add a "learning about Mirage" page

### PR DESCRIPTION
**Pull request by mk270**

### CODE REVIEW

**Comment by hannesm** [comment]
thanks for this contribution, since @mato @djwillia and @ricarkol wrote the solo5 text, and I'm not sure whether that should be hosted on mirage.io or somewhere on solo5 (maybe in its README.md?), and also I'm not sure they're happy with that text being public _right now_, I mention them here so they can discuss and decide.

**Comment by mk270** [comment]
Ok, I did ask @mato, but let's double check.

**Comment by mato** [comment]
@mk270 The text is a bit future-looking, as you can see yourself where you write a bunch of stuff in the following paragraphs describing the actual current status :-) Perhaps have a go at editing it down to reflect the current status more in your own words, or would you like me to do that?

Its final home will probably be the Solo5 README.md. but, as @hannesm mentioned I'd prefer to leave the future-looking stuff out of it for now (since otherwise people will go looking for it and get confused).

**Comment by mk270** [comment]
Fair enough - this is an object lesson in why pull requests should cover exactly one topic. My bad.

**Comment by mato** [comment]
I'd say keep this PR to just the `learning.md` that you've added (which looks good!) and write the minimal amount of background about the backends in your own words, linking to the Solo5 README.md if you want.

**Comment by mk270** [comment]
how's that?

**Comment by mato** [comment]
@mk270 TBH I'd just drop the Solo5 page entirely as it still kind of contradicts itself with the sandbox-agnostic stuff which we don't have yet. Just to the Github repo for now if you don't mind.

The core of learning.md as you've written seems fine to me, thanks!

File: files/css/site.css [hunk 1]
```diff
@@ -2,6 +2,7 @@ h1 { font-size: 2.2rem; font-weight: bold; }
 h2 { font-size: 1.8rem; font-weight: bold; }
 h3 { font-size: 1.6rem; font-weight: 600; }
 h4,h5,h6 { color: #222222; font-size: 1.4rem; font-weight: 600; }
+h1,h2,h3,h4,h5,h6 { margin-top: 2rem; }
```
Comment (mato) [inline files/css/site.css]: Nit: Not sure what this is changing w/o actually looking at the result, but if it is what I think it is (vertical margins should match line height) then you want the `margin-top` folded into the existing `h1 ... h6` above and use a `margin-top` equal to the appropriate `font-size`.

**Comment by mato** [comment]
@mk270 This looks good to me. However, I don't know how it is intended to fit in with the existing material on the website, e.g. the "Getting Started" section at https://mirage.io/docs/. Not sure who can answer this?

**Comment by mk270** [comment]
It's intended to be linked from that page, and give a higher-level overview of the learning curve.

**Comment by mato** [comment]
Ok, good from my PoV. Will leave this for @hannesm or others to merge if they're happy with it.

File: tmpl/wiki/learning.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+* `ukvm` (part of [Solo5](https://github.com/Solo5/solo5)
+* `xen`
+
+The backends above are listed in ascending order of invasiveness. `unix` runs
+as an unprivileged process on your unmodified Linux kernel. `ukvm` uses
+Solo5, typically in a KVM virtual machine on Linux, which entails some setup
+work, e.g., of IP routing. `xen` requires that Xen be run underneath your
+operating system(s), and that Mirage will be run directly on top of Xen.
+Installing Xen is not hard (about 20 minutes), but your host machine must
+then be dedicated to running Xen.
```
Comment (yomimono) [inline tmpl/wiki/learning.md]: I think "your host machine must then be dedicated to running Xen" is too big a statement; my non-Qubes Xen machine dual-boots Xen with a Linux dom0 and that Linux directly without Xen, and that's the default configuration for `apt-get install xen-hypervisor` in Ubuntu.

File: tmpl/wiki/index.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ MirageOS is in beta stage and the infrastructure you see here is self-hosting. T
 <ul>
 <li><a href="/wiki/install">Installation</a></li>
 <li><a href="/wiki/hello-world">Hello Mirage World</a></li>
+<li><a href="/wiki/learning">Learning about Mirage</a></li>
```
Comment (yomimono) [inline tmpl/wiki/index.md]: I wonder whether this should be listed above Hello Mirage World (but I don't feel strongly about it).

**Comment by yomimono** [comment]
A few nits noted.  Thanks for putting this document together, @mk270 ! :)

File: tmpl/wiki/learning.md (hunk 2 above)
Comment (mk270) [reply tmpl/wiki/learning.md]: yeah, and that was the watered down version; I shall water it down some more!

**Comment by mk270** [comment]
Ok, @hannesm please take another look

File: src/data.ml [hunk 4]
```diff
@@ -597,6 +604,21 @@ module Wiki = struct
     weekly ~y:2014 ~m:3 ~d:18 ~a:amir;
     weekly ~y:2014 ~m:3 ~d:4 ~a:amir;
     weekly ~y:2014 ~m:2 ~d:26 ~a:amir;
+
+    { updated    = date (2017, 12, 4, 15, 00);
+      author     = mk270;
+      subject    = "Solo5 backend";
+      body       = File "backend-solo5.md";
+      permalink  = "backend-solo5";
+    };
```
Comment (hannesm) [inline src/data.ml]: that's now superfluous, no?

File: tmpl/wiki/learning.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+## What are backends?
+
+Backends include:
+
+* `unix`
+* `ukvm` (part of [Solo5](https://github.com/Solo5/solo5)
+* `xen`
+
+The backends above are listed in ascending order of invasiveness. `unix` runs
+as an unprivileged process on your unmodified Linux kernel. `ukvm` uses
```
Comment (hannesm) [inline tmpl/wiki/learning.md]: this is only partially true, `unix` with a `direct` network stack needs root privileges to open a `tap` network device

**Comment by hannesm** [comment]
I like the text, but there's quite a bit of overlap with https://mirage.io/community/ (which mentions irc channel and mailing list, but not discuss.ocaml.org).  The issue I have with this is that duplicated information gets usually outdated quickly.  This is related to https://github.com/mirage/mirage/issues/569 and I suspect there's maybe a redesign of the information on the website needed.

I don't know, and let others decide whether to merge this.  I won't have time to (re)design the website and won't be willing to be in charge for that, sorry.

**Comment by samoht** [comment]
LGTM module @hannesm's remark on the superfluous entry. I am happy to merge tomorrow if no one else did it before me.

File: tmpl/wiki/learning.md (hunk 5 above)
Comment (mk270) [reply tmpl/wiki/learning.md]: amended

**Comment by samoht** [comment]
@mk270 I meant [this](https://github.com/mirage/mirage-www/pull/577/files#r156503414) comment about the `"Solo5 backend"` entry.

**Comment by samoht** [comment]
thanks!

**Pull request merged**